src/vmm/src/arch/x86_64/layout.rs

```diff
@@ -29,3 +29,41 @@ pub const KVM_TSS_ADDRESS: u64 = 0xfffb_d000;
 
 /// The 'zero page', a.k.a linux kernel bootparams.
 pub const ZERO_PAGE_START: u64 = 0x7000;
+
+/// APIC address
+pub const APIC_ADDR: u32 = 0xfee0_0000;
+
+/// IOAPIC address
+pub const IOAPIC_ADDR: u32 = 0xfec0_0000;
+
+/// Location of RSDP pointer in x86 machines
+pub const RSDP_ADDR: u64 = 0x000e_0000;
+
+/// Start of memory region we will use for system data (MPTable, ACPI, etc). We are putting it
+/// where EBDA normally is, i.e. in the last 1 KiB of the first 640KiB of memory, i.e in the range:
+/// [0x9FC00, 0x9FFFF]
```

Comment:
I'm a bit confused by this comment. I know it originates from the MPTable code, so I think its simply outdated (below we say that we resert 257KiB for ACPI and friends, so the "we are putting it in the [1KiB] range" seems wrong now), but then below you say that even for MPTables we can use up to 5KB, so how was this comment ever true? D:

Reply:
You are right. It should say: "We are putting _starting from_ where EBDA normally is, i.e. ...", as in, the beginning of this area is there but now it extends to more than that. Will remove the range as well.

What I did here is reserve the EBDA area plus a bit more now, all the way up to (but not included) the `HIMEM_START`. We do not need all of this, but I'm just reserving the space for "future use".

> but then below you say that even for MPTables we can use up to 5KB, so how was this comment ever true?

That was true because we limit the number of vCPUs to 32. Both specs use as apic id a `u8` though, which implies up to 256 vCPUs. With this limit we would need ~5000 bytes for describing vCPUS in MPTable :)

Reply:
Ahh, right the 32 vcpu limit, thanks for the explaination! :D